src/parser/creative_companion_parser.js

```diff
@@ -14,79 +14,89 @@ import { parserUtils } from './parser_utils';
  */
 export function parseCreativeCompanion(creativeElement, creativeAttributes) {
   const creative = new CreativeCompanion(creativeAttributes);
+  creative.required = creativeElement.getAttribute('required') || null;
 
-  parserUtils
+  creative.variations = parserUtils
     .childrenByName(creativeElement, 'Companion')
-    .forEach(companionResource => {
-      const companionAd = new CompanionAd();
-      companionAd.id = companionResource.getAttribute('id') || null;
-      companionAd.width = companionResource.getAttribute('width');
-      companionAd.height = companionResource.getAttribute('height');
-      companionAd.companionClickTrackingURLTemplates = [];
+    .map(companionResource => {
+      const attributes = {};
+      const companionAttrs = companionResource.attributes;
+      for (let i = 0; i < companionAttrs.length; i++) {
+        attributes[companionAttrs[i].nodeName] = companionAttrs[i].nodeValue;
+      }
+      const companionAd = new CompanionAd(attributes);
```

Comment:
I like this approach of passing at least the attributes to the constructor (and the class logic will take care of assigning them to the right properties).

Maybe we can factor this in an util method in order to use it in the future too, something like `getAttributes`:
```javascript
function getAttributes(node) {
  const attributes = {};
  const nodeAttributes = node.attributes;

  for (let i = 0; i < companionAttrs.length; i++) {
    attributes[nodeAttributes[i].nodeName] = nodeAttributes[i].nodeValue;
  }

  return attributes
}
```

Reply:
created function parseAttributes